### DISCUSSION TITLE
How to increase the DRAM space?

### DISCUSSION THREAD

**Discussion by amax-chang** [QQxiaoming/quard_star_tutorial#24] [Q&A] [answered]
I've tried to increase DRAM space by modifying dts/quard_star_uboot.dts: add extra entries to "reg" of "memory".
But it failed to get more RAM under Linux, still the same 768 MB...
What key modifications are need to increase the usable RAM space to above 32-bit limit, for example 8 GB? 
Thanks!

**Comment by QQxiaoming** [comment, marked as answer]
Thank you for your question. I have tested that it is completely feasible to configure more dram sizes. You only need to do the following two configurations:

1. Increase the memory address (starting from 0x1_0000_0000) in both quard_star_sbi.dts and quard_star_uboot.dts files.
2. Modify `-m 1G \` in run.sh. https://github.com/QQxiaoming/quard_star_tutorial/blob/3e57a712354e7526695fa4abff5d8db6d20847c8/run.sh#L212

**Reply by amax-chang** [reply]
I've tried:
1. "memory" of quard_star_uboot.dts is modified as:

        memory@80000000 {
                device_type = "memory";
                reg = <0x0 0x80000000 0x0 0x3f800000>,
                      <0x1 0x00000000 0x0 0x80000000>;
        };

2. quard_star_sbi.dts is modified as:

        memory@80000000 {
                device_type = "memory";
                reg = <0x00000000 0x80000000 0x00000000 0x80000000>,
                      <0x00000001 0x00000000 0x00000000 0x80000000>;
        };

3. execute "./build.sh all" to update the images

4. modify "-m 1G \" to in run.sh to "-m 8G'

5. run it: "./run.sh nographic"

6. after boot and login, the output of "cat /proc/meminfo":
MemTotal:         765116 kB
MemFree:          708528 kB
MemAvailable:     715808 kB
... 

Any suggestions?
Thanks!

**Reply by QQxiaoming** [reply]
> 3. execute "./build.sh all" to update the images

Here you should use `./build.sh all bootfs` instead of `./build.sh all`, because the dtb and zImage files in the file system and the rootfs.img are not updated by default.

BUILD_ROOTFS_OPT has three options: `all`, `bootfs` and others. The default is other, and the message "`skip build rootfs.img!`" will be output.

https://github.com/QQxiaoming/quard_star_tutorial/blob/9ffd29c32523fa8976362ff9b16082d90b9a758e/build.sh#L281-L347

Admittedly, the design here may not be very good, which brings confusion, and I am also thinking about how to modify this script.

**Reply by amax-chang** [reply]
After "./build.sh all bootfs" and restart, I got "MemTotal: 2741956 kB".
Yes it works!

It is great that you keep active to update this project...
Thanks a lot!
